sim/battle-actions.ts

```diff
@@ -313,6 +313,8 @@ export class BattleActions {
 			}
 		}
 		if (noLock && pokemon.volatiles['lockedmove']) delete pokemon.volatiles['lockedmove'];
+		this.battle.faintMessages();
+		this.battle.checkWin();
```

Comment:
Wait, I'm confused. `faintMessages` still runs `checkWin` when invoked like this, so the second `checkWin` is useless, right?

Reply:
If the `faintMessages` mid-attack fainted the target, there's nothing in the faint queue when it gets here. Without running `checkWin` separately, the first test I added fails because the battle isn't actually over.